ultralytics/utils/ops.py

```diff
@@ -848,17 +848,18 @@ def clean_str(s):
     """
     return re.sub(pattern="[|@#!¡·$€%&()=?¿^*;:,¨´><+]", repl="_", string=s)
 
+
 def v10postprocess(preds, max_det, nc=80):
-    assert(4 + nc == preds.shape[-1])
+    assert (4 + nc == preds.shape[-1])
     boxes, scores = preds.split([4, nc], dim=-1)
     max_scores = scores.amax(dim=-1)
-    max_scores, index = torch.topk(max_scores, max_det, axis=-1)
+    max_scores, index = torch.topk(max_scores, max_det, dim=-1)
     index = index.unsqueeze(-1)
     boxes = torch.gather(boxes, dim=1, index=index.repeat(1, 1, boxes.shape[-1]))
     scores = torch.gather(scores, dim=1, index=index.repeat(1, 1, scores.shape[-1]))
 
-    scores, index = torch.topk(scores.flatten(1), max_det, axis=-1)
-    labels = index % nc
+    scores, index = torch.topk(scores.flatten(1), max_det, dim=-1)
+    labels = (index % nc).to(torch.float32)
```

Comment:
We previously added the cast for labels here
https://github.com/THU-MIG/yolov10/blob/13f6ab770d561e715a094f94594ca34b80dba1ee/ultralytics/nn/modules/head.py#L523 Could you please also revert this change? Thanks!